### PR TITLE
Ensure controller manager and scheduler can perform delegated auth checks

### LINKED ISSUE #71752: kube-scheduler is missing permissions for delegated authentication
**What happened**:
The kube-scheduler crashes with v1.13 as it is missing the required permissions for delegated authentication:
```
W1205 10:35:37.728989       1 authentication.go:248] Unable to get configmap/extension-apiserver-authentication in kube-system.  Usually fixed by 'kubectl create rolebinding -n kube-system ROLE_NAME --role=extension-apiserver-authentication-reader --serviceaccount=YOUR_NS:YOUR_SA'
configmaps "extension-apiserver-authentication" is forbidden: User "system:kube-scheduler" cannot get resource "configmaps" in API group "" in the namespace "kube-system"
```

kube-scheduler runs in a different kubernetes cluster.

v1.12 does not have the `--authentication-kubeconfig` flag - and it does not try to load the ConfigMap kube-system/extension-apiserver-authentication

v1.13 without `--authentication-kubeconfig` uses the in-cluster client - pointing to the wrong cluster & fails loading the ConfigMap kube-system/extension-apiserver-authentication

v1.13 with `--authentication-kubeconfig` uses the provided kubeconfig, but still fails loading the ConfigMap kube-system/extension-apiserver-authentication as the user `system:kube-scheduler` does not have the required permission to read it.

**What you expected to happen**:
1. The kube-scheduler has the correct permission by default to handle delegated authentication
2. The kube-scheduler handles this case gracefully by not serving on the secure port. See https://github.com/kubernetes/kubernetes/pull/71743#issuecomment-444561450

**How to reproduce it (as minimally and precisely as possible)**:
Start a v1.13.0 kube-scheduler (Not inside the cluster) with the following flags:
```
--kubeconfig /etc/kubernetes/kubeconfig/kubeconfig
--authentication-kubeconfig /etc/kubernetes/kubeconfig/kubeconfig
```

**Anything else we need to know?**:
The environment:
kube-scheduler runs in a different kubernetes cluster.

Introduced in #69663
This is a tracking issue triggered by: https://github.com/kubernetes/kubernetes/pull/71743

**Environment**:
- Kubernetes version (use `kubectl version`): v1.13.0

/kind bug

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
* Includes permissions required to perform delegated authn/authz checks in controller manager and scheduler roles
* Grants permissions required to read cluster authentication configuration to controller manager and scheduler

**Which issue(s) this PR fixes**:
Fixes #71752

**Does this PR introduce a user-facing change?**:
```release-note
`system:kube-controller-manager` and `system:kube-scheduler` users are now permitted to perform delegated authentication/authorization checks by default RBAC policy
```

### CODE REVIEW

**Comment by sttts** [comment]
Can these go? https://github.com/kubernetes/kubernetes/blob/1588af4031e746e3e48855256054133e8707f018/test/integration/serving/serving_test.go#L115

File: plugin/pkg/auth/authorizer/rbac/bootstrappolicy/namespace_policy.go [hunk 1]
```diff
@@ -119,10 +120,15 @@ func init() {
 			rbacv1helpers.NewRule("get", "update").Groups(legacyGroup).Resources("configmaps").Names("kube-scheduler").RuleOrDie(),
 		},
 	})
+
+	delegatedAuthBinding := rbacv1helpers.NewRoleBinding("extension-apiserver-authentication-reader", metav1.NamespaceSystem).Users(user.KubeControllerManager, user.KubeScheduler).BindingOrDie()
+	delegatedAuthBinding.Name = "system::extension-apiserver-authentication-reader"
```
Comment (sttts) [inline plugin/pkg/auth/authorizer/rbac/bootstrappolicy/namespace_policy.go]: why not the same style as below?

Comment (liggitt) [reply plugin/pkg/auth/authorizer/rbac/bootstrappolicy/namespace_policy.go]: wanted all bindings we create to start with `system:`... by default, this copies the name of the role

**Comment by liggitt** [comment]
> Can these go?

yup, switched that to use the standard controller manager username and removed all custom bindings from the test

**Pull request merged**